rails/app/controllers/api/communities_controller.rb

```diff
@@ -0,0 +1,18 @@
+module Api
+  class CommunitiesController < BaseController
+    def index
+      @communities = Community.where(public: true).order(updated_at: :desc)
+      @communities = @communities.where("name ilike :search", search: "%#{params[:search]}%") if params[:search]
```

Comment:
Agreed! Since it's behind a feature flag at the moment, I suspect we won't have too many at the moment, so this should be fine. That said, using pg text search would allow partial matches, which could be useful. Either way, it's probably out of scope of this PR and can be logged as a future enhancement.